numpy/_core/tests/test_deprecations.py

```diff
@@ -90,7 +90,7 @@ def assert_deprecated(self, function, num=1, ignore_others=False,
 
         try:
             function(*args, **kwargs)
-        except (Exception if function_fails else ()):
+        except (Exception if function_fails else ()):  # noqa: B030
```

Comment:
It might be cleaner to instead use `contextlib.raises` and `contextlib.nullcontext`. There's also a ruff rule for this if I remember correctly.